larpandoracontent/LArObjects/LArAdaBoostDecisionTree.h

```diff
@@ -0,0 +1,319 @@
+         */
+        double GetThreshold() const;
+
+        /**
+         *  @brief  Return cut variable
+         */
+        int GetVariableID() const;
+
+        /**
+         *  @brief  Return outcome
+         */
+        bool GetOutcome() const;
+
+    private:
+        const int       m_nodeID;               ///< Node id
+        const int       m_parentNodeID;         ///< Parent node id
+        const int       m_leftChildNodeID;      ///< Left child node id
+        const int       m_rightChildNodeID;     ///< Right child node id
+        const bool      m_isLeaf;               ///< Is node a leaf
+        const double    m_threshold;            ///< Threshold used for decision if decision node
+        const int       m_variableID;          ///< Variable cut on for decision if decision node
+        const bool      m_outcome;              ///< Outcome if leaf node
+    };
+
+    typedef std::map<int, const Node*> IDToNodeMap;
+
+    class WeakClassifier
+    {
+    public: 
+        /**
+         *  @brief  Constructor, set hierarchy for nodes 
+         */
+        WeakClassifier(IDToNodeMap &idToNodeMap, const double &weight, const int &treeID);
+
+        /**
+         *  @brief  Predict signal or background based on trained data
+         *
+         *  @param  features the input features 
+         */
+        bool Predict(const DoubleVector &features) const;
+
+        /**
+         *  @brief  Evalute node and return outcome
+         *
+         *  @param  nodeID current node id 
+         *  @param  features the input features 
+         */
+        bool EvaluateNode(const int &nodeID, const DoubleVector &features) const;
+
+        /**
+         *  @brief  Get boost weight for weak classifier
+         */
+        double GetWeight() const;
+
+        /**
+         *  @brief  Get tree id for weak classifier
+         */
+        int GetTreeID() const;
+
+    private:
+        IDToNodeMap     m_idToNodeMap; ///< Decision tree nodes
+        const double    m_weight;      ///< Boost weight 
+        const int       m_treeID;      ///< Decision tree id        
+    };
+
+    typedef std::vector<const WeakClassifier*> WeakClassifiers;
+
+    class StrongClassifier
+    {
+    public:
+        /**
+         *  @brief  Constructor
+         */
+        StrongClassifier(WeakClassifiers &weakClassifiers);
+
+        /**
+         *  @brief  Predict signal or background based on trained data
+         *
+         *  @param  features the input features 
+         */
+        double Predict(const DoubleVector &features) const;
+
+    private: 
+        WeakClassifiers     m_weakClassifiers;     ///< Vector of weak classifers 
+    };
+
+    /**
+     *  @brief  Read the component at the current xml element
+     *
+     *  @param  pCurrentXmlElement address of the current xml element
+     *
+     *  @return success
+     */
+    pandora::StatusCode ReadComponent(pandora::TiXmlElement *pCurrentXmlElement);
```

Comment:
Check whether can be const member functions, as per earlier comment.

Reply:
Done.